common/src/allocations/monitor.rs

```diff
@@ -183,16 +147,41 @@ mod test {
     async fn test_parses_allocation_data_from_network_subgraph_correctly() {
         let (network_subgraph, _mock_server) = setup_mock_network_subgraph().await;
 
-        let allocations = indexer_allocations(
+        let result = get_allocations(
             network_subgraph,
             *test_vectors::INDEXER_ADDRESS,
-            Duration::from_secs(60),
             Duration::from_secs(300),
-        );
+        )
+        .await;
+        match result {
+            Ok(allocations) => {
+                assert_eq!(allocations, *test_vectors::INDEXER_ALLOCATIONS)
+            }
+            Err(e) => println!("Error: {:#?}", e),
+        }
+    }
 
-        assert_eq!(
-            allocations.value().await.unwrap(),
-            *test_vectors::INDEXER_ALLOCATIONS
-        );
+    fn network_subgraph_client() -> &'static SubgraphClient {
+        Box::leak(Box::new(SubgraphClient::new(
+            reqwest::Client::new(),
+            None,
+            DeploymentDetails::for_query_url(NETWORK_SUBGRAPH_URL).unwrap(),
+        )))
+    }
+
+    #[tokio::test]
+    async fn test_network_query() {
+        let result = get_allocations(
+            network_subgraph_client(),
+            Address::from_str("0xa7a068c169d3f38efc62a1d6ff8de8b1ddfc8a57").unwrap(),
+            Duration::from_secs(1712448507),
+        )
+        .await;
+        match result {
+            Ok(allocations) => {
+                assert!(allocations.len() > 1)
+            }
+            Err(e) => println!("Error: {:#?}", e),
+        }
     }
```

Comment:
Speaking of which, in case there's an error it's not failing.

Reply:
Even if they are closed they should still appear.. or is there a system that deletes them? 
But yeah depending on external source sucks but not sure how else to really test this is doing what its intended without mocking everything which then it would lose all meaning

Reply:
@carlosvdr oh yeah you're right, what makes it guaranteed to return stuff is the `1712448507` timestamp. In that case you can improve that test in 2 ways:
- Make the pages shorter, 1000 -> 200. Not really a test improvement, but I think it's a more reasonable default value.
- Find an indexer / timestamp pair that yields much more than 200 allocations. Since the number of allocations can only increase over time (because the timestamp is hardcoded), you can test like `assert!(allocations.len() > 1234)`